protocols/imap/src/main/java/org/apache/james/imap/processor/SearchProcessor.java

```diff
@@ -262,16 +262,14 @@ private ModSeq findHighestModSeq(MailboxSession session, MessageManager mailbox,
         return highestModSeq;
     }
 
-
-    private SearchQuery toQuery(SearchKey key, ImapSession session) throws MessageRangeException {
-        final SearchQuery result = new SearchQuery();
-        final SelectedMailbox selected = session.getSelected();
+    private SearchQuery.Builder toQuery(SearchKey key, ImapSession session) throws MessageRangeException {
+        SearchQuery.Criterion criterion = toCriterion(key, session);
+        SearchQuery.Builder builder = SearchQuery.builder();
+        SelectedMailbox selected = session.getSelected();
         if (selected != null) {
-            result.addRecentMessageUids(selected.getRecent());
+            builder.addRecentMessageUids(selected.getRecent());
         }
-        final SearchQuery.Criterion criterion = toCriterion(key, session);
-        result.andCriteria(criterion);
-        return result;
+        return builder.andCriteria(criterion);
```

Comment:
why not building here and returning a SearchQuery instead of a SearchQuery.Builder?